src/MailViewer.php

```diff
@@ -42,9 +42,20 @@ public static function find(string $mail)
                 $args = [];
 
                 foreach ($dependencies as $dependency) {
+                    $factoryStates = [];
+
+                    if (is_array($dependency)) {
+                        // Check if the dependency array contains factory states
+                        if (in_array('states', array_keys($dependency)) === true) {
+                            // Set the states and actual dependency class
+                            $factoryStates = $dependency['states'];
+                            $dependency = $dependency['class'];
+                        }
+                    }
+
                     if (is_string($dependency) && class_exists($dependency)) {
                         if (isset($eloquentFactory[$dependency])) {
-                            $args[] = factory($dependency)->create();
+                            $args[] = factory($dependency)->states($factoryStates)->make();
```

Comment:
Curious to know why `make()` over `create()` 😄

Reply:
The first reason was the tests, the create function will try to save the Test model in the database as it's a Illuminate model. I can make it some general model that doesn't save to the database to make the test pass too. The other reason is that you probably don't want to save the records to the database when you are just viewing the email via the `/mails` page, especially on a production environment ;) I actually didn't realise this was happening until I was editing this code.